### PR TITLE
Add modeling rule schema validation

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready
- [x] On hold - until the next sdk release

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3258

#### Modeling Rule Validations

 - New validation to check if the schema file (.json) exists for a modeling rule.
 - Validation to verify that the `rules` and `schema` keys in the yml file are empty (empty string)
 - Validation to validate that the files names match the folder name

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/modeling_rule.py [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
     def is_valid_file(self, validate_rn=True, is_new_file=False, use_git=False):
         """
         Check whether the modeling rule is valid or not
         Note: For now we return True regardless of the item content. More info:
         https://github.com/demisto/etc/issues/48151#issuecomment-1109660727
         """
-        logging.debug('Automatically considering XSIAM content item as valid, see issue #48151')
-        return True
+
+        self.is_schema_file_exists()
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/hook_validations/modeling_rule.py:29]: That's good, let's also add a function that checks that the schema key in the yml itself is empty, so everyone will work with the new standard.
```suggestion
        self.is_schema_file_exists()
        self.is_schema_key_empty()
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -519,7 +522,7 @@ def file_type_not_supported():
         return "The file type is not supported in the validate command.\n" \
                "The validate command supports: Integrations, Scripts, Playbooks, " \
                "Incident fields, Incident types, Indicator fields, Indicator types, Objects fields, Object types," \
-               " Object modules, Images, Release notes, Layouts, Jobs, Wizards, and Descriptions."
+               " Object modules, Images, Release notes, Layouts, Jobs, Wizards, Descriptions And Modeling Rules."
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/errors.py:525]: nice

File: demisto_sdk/commands/common/tests/modeling_rules_test.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import os
+
+from demisto_sdk.commands.common.hook_validations.modeling_rule import \
+    ModelingRuleValidator
+from demisto_sdk.commands.common.hook_validations.structure import \
+    StructureValidator
+from TestSuite.test_tools import ChangeCWD
+
+
+def test_is_valid_modeling_rule(repo):
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/tests/modeling_rules_test.py:10]: Good test, let's add doc string and given when then.

File: demisto_sdk/commands/common/tests/modeling_rules_test.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    pack = repo.create_pack('TestPack')
+    dummy_modeling_rule = pack.create_modeling_rule('MyRule')
+    structure_validator = StructureValidator(dummy_modeling_rule.yml.path)
+    with ChangeCWD(repo.path):
+        modeling_rule_validator = ModelingRuleValidator(structure_validator)
+        assert modeling_rule_validator.is_schema_file_exists()
+
+
+def test_is_invalid_modeling_rule(repo):
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/tests/modeling_rules_test.py:19]: Good test, let's add doc string and given when then.

File: demisto_sdk/commands/common/tests/modeling_rules_test.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    pack = repo.create_pack('TestPack')
+    dummy_modeling_rule = pack.create_modeling_rule('MyRule')
+    structure_validator = StructureValidator(dummy_modeling_rule.yml.path)
+    schema_path = dummy_modeling_rule.schema.path
+    if os.path.exists(schema_path):
+        os.remove(schema_path)
+
+    with ChangeCWD(repo.path):
+        modeling_rule_validator = ModelingRuleValidator(structure_validator)
+        assert not modeling_rule_validator.is_schema_file_exists()
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/tests/modeling_rules_test.py:29]: Let's add another negative test for invalid modeling that will check that when modeling with a non-empty schema field is given the validation fails.

**Review by ShahafBenYakir — COMMENTED** [review]
NIce.
Will your previous validation about the naming convention (entities name is the same as the folder containing it) also catch modeling rules? If not what is the effort of adding it?

File: demisto_sdk/commands/common/hook_validations/modeling_rule.py (hunk 1 above)
Comment (Shellyber) [reply demisto_sdk/commands/common/hook_validations/modeling_rule.py:29]: In that effort, would you like me to check that the `rules` key in the yml also empty?

**Comment by Shellyber** [comment]
@ShahafBenYakir Done. and added the naming validation also

File: demisto_sdk/commands/split/ymlsplitter.py [hunk 6]
```diff
@@ -118,9 +119,9 @@ def extract_to_package_format(self) -> int:
             self.extract_rules(f'{output_path}/{base_name}.xif')
             self.extract_rule_schema_and_samples(f'{output_path}/{base_name}.json')
             if 'rules' in yaml_obj:
-                yaml_obj['rules'] = ''
+                yaml_obj['rules'] = PlainScalarString('')
```
[the remark this answers is not in the document]
Comment (Shellyber) [reply demisto_sdk/commands/split/ymlsplitter.py:122]: Just a small fix - I found a way to put an empty string instead of `>`
So now it will look like this:
```
rules: ''
schema: ''
```

[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply demisto_sdk/commands/split/ymlsplitter.py:122]: nice

**Review by ShahafBenYakir — APPROVED** [review]
